### PR TITLE
First Criteria Iteration

### PR DESCRIPTION
**Pull request by paulktham**
implementation of a criteria requirement, more details at
https://github.com/orgs/AY2425S1-CS2113-W12-2/projects/1/views/1?pane=issue&itemId=84134688&issue=AY2425S1-CS2113-W12-2%7Ctp%7C52

### CODE REVIEW

File: src/main/java/allocator/Allocator.java [hunk 1]
```diff
@@ -5,20 +5,45 @@
...
     
+    private static double minimumGPA = 0;
     private StudentList studentList;
+    
 
     public Allocator(StudentList studentList) {
         this.studentList = new StudentList(studentList);  // don't want to modify the original list
     }
+
+    public void setMinimumGPA(String stringGpa) throws SEPException{
```
Comment (isaacsaw25) [inline src/main/java/allocator/Allocator.java:24]: Oopsie missing a space here after `SEPException`

File: src/main/java/command/CriteriaCommand.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import exceptions.SEPException;
+import studentlist.StudentList;
+import ui.UI;
+
+public class CriteriaCommand extends Command {
+    private String input;
+    private UI ui;
+    private Allocator allocator;
+
+    public CriteriaCommand(StudentList studentList,String input, UI ui) {
```
Comment (isaacsaw25) [inline src/main/java/command/CriteriaCommand.java:13]: Similarly, a space after `StudentList studentList,` and before `String input`

File: src/main/java/allocator/Allocator.java [hunk 3]
```diff
@@ -5,20 +5,45 @@
...
+        return minimumGPA;
+    }
     
     public StudentList allocate() {
         studentList.sortStudentsByGPA();
         for (Student student : studentList.getList()) {
             for (int uni : student.getUniPreferences()) {
                 University university = UniversityRepository.getUniversityByIndex(uni);
-                if (university.getSpotsLeft() > 0) {
+                if (university.getSpotsLeft() > 0 && student.getGpa() >= minimumGPA) {
```
Comment (isaacsaw25) [inline src/main/java/allocator/Allocator.java:46]: My idea of minimumGpa was for it to be a member of each `University` object, but at the moment setting the criteria as a `Allocator` level variable seems to be fine as well. 👍

**Review by isaacsaw25 — CHANGES_REQUESTED** [review]
Do fix the spaces 🙏 🙏

File: src/main/java/allocator/Allocator.java (hunk 1 above)
Comment (paulktham) [reply src/main/java/allocator/Allocator.java]: Good catch!

File: src/main/java/command/CriteriaCommand.java (hunk 2 above)
Comment (paulktham) [reply src/main/java/command/CriteriaCommand.java]: Good catch!

File: src/main/java/allocator/Allocator.java (hunk 3 above)
Comment (paulktham) [reply src/main/java/allocator/Allocator.java:53]: Thanks!

**Review by isaacsaw25 — APPROVED** [review]

**Review by ThisisXXZ — APPROVED** [review]
Nice work implementing new commands!

**Pull request merged**